### ISSUE TITLE
Prior::get_value() is still `float`

### ISSUE THREAD

**Issue by KrisThielemans** [SyneRBI/SIRF#1290]
`stir::GeneralisedPrior::compute_value() `calculations are using and returning `double`. We currently return this after casting to `float`
https://github.com/SyneRBI/SIRF/blob/1e2c5c125df4216fc7bf593047f6056d94df3ae5/src/xSTIR/pSTIR/STIR.py#L2323
https://github.com/SyneRBI/SIRF/blob/1e2c5c125df4216fc7bf593047f6056d94df3ae5/src/xSTIR/cSTIR/cstir.cpp#L1332-L1333

A similar problems occured for the log-likelihood but was recently fixed.

**Comment by evgueni-ovtchinnikov** [comment]
Is it not a STIR issue? - see `recon_buildblock/GeneralisedPrior.h`:
```
template <typename DataT>
class GeneralisedPrior : public RegisteredObject<GeneralisedPrior<DataT>>
{
public:
  inline GeneralisedPrior();

  //! compute the value of the function
  /*! For derived classes where this doesn't make sense, it's recommended to return 0.
   */
  virtual double compute_value(const DataT& current_estimate) = 0;
```

**Comment by KrisThielemans** [comment]
why? STIR returns `double`, SIRF casts to `float`.

**Comment by evgueni-ovtchinnikov** [comment]
I thought you wanted float, sorry, fixed now by #1291.
